tool/planet/etcd.go

```diff
@@ -179,45 +195,42 @@ func etcdDisable(upgradeService, stopAPIServer bool) error {
 		}
 	}
 
-	if upgradeService {
-		return trace.Wrap(disableService(ctx, ETCDUpgradeServiceName))
-	}
-
 	return trace.Wrap(disableService(ctx, ETCDServiceName))
 }
 
-// etcdEnable enables a disabled etcd node
-func etcdEnable(upgradeService bool, joinToMaster string) error {
-	ctx, cancel := context.WithTimeout(context.Background(), EtcdUpgradeTimeout)
-	defer cancel()
-
-	if !upgradeService {
-		ctx, cancel := context.WithTimeout(context.Background(), 120*time.Minute)
-		defer cancel()
-		err := etcdInitJoin(ctx, joinToMaster)
-		if err != nil {
-			return trace.Wrap(err)
-		}
-
-		// restart the clients of the etcd service when the etcd service is brought online, which usually will be post
-		// upgrade. This will ensure clients running inside planet are restarted, which will refresh any local state
-		restartEtcdClients(ctx)
-		return trace.Wrap(enableService(ctx, ETCDServiceName))
-	}
+// etcdEnableUpgrade enables a disabled etcd upgrade service
+func etcdEnableUpgrade() error {
 	// don't actually enable the service if this is a proxy
 	env, err := box.ReadEnvironment(ContainerEnvironmentFile)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	if env.Get(EnvEtcdProxy) == EtcdProxyOn {
-		log.Info("etcd is in proxy mode, nothing to do")
+		log.Info("Etcd is in proxy mode, nothing to do.")
 		return nil
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), EtcdUpgradeTimeout)
+	defer cancel()
 	return trace.Wrap(enableService(ctx, ETCDUpgradeServiceName))
 }
 
+// etcdEnable enables a disabled etcd node
+func etcdEnable(joinToMaster string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Minute)
+	defer cancel()
+	err := etcdInitJoin(ctx, joinToMaster)
```

Comment:
If we just use existing data dirs on all nodes, we won't need to join anything anywhere, no?

Reply:
Yeah, makes sense to remove.